pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -73,7 +78,7 @@ public BaseFilterOperator getLeafFilterOperator(QueryContext queryContext, Predi
       if (predicateEvaluator.isAlwaysFalse()) {
         return EmptyFilterOperator.getInstance();
       } else if (predicateEvaluator.isAlwaysTrue()) {
-        return new MatchAllFilterOperator(numDocs);
+        return new MatchAllFilterOperator(queryContext, dataSource, numDocs);
```

Comment:
I'd suggest handling it here instead of within the `MatchAllFilterOperator`. If there is null values, we can return a `BitmapBasedFilterOperator`. Handling it within `MatchAllFilterOperator` will break the `isResultMatchingAll()` contract

Reply:
Your suggestion is a cleaner solution.

Done.